src/tests/end-to-end/common/page-controllers/page.ts

```diff
@@ -278,26 +279,17 @@ export class Page {
         }
     }
 
-    private async screenshotOnError<T>(fn: () => Promise<T>): Promise<T> {
-        try {
-            return await fn();
-        } catch (originalError) {
-            try {
-                await takeScreenshot(this.underlyingPage);
-            } catch (secondaryTakeScreenshotError) {
-                console.error(
-                    `screenshotOnError: Detected an error, and then *additionally* hit a second error while trying to take a screenshot of the page state after the first error.\n` +
-                        `screenshotOnError: The secondary error from taking the screenshot is: ${secondaryTakeScreenshotError.stack}\n` +
-                        `screenshotOnError: rethrowing the original error...`,
-                );
-            }
-            throw originalError;
-        }
-    }
-
     public async injectScriptFile(filePath: string): Promise<void> {
         await this.screenshotOnError(async () => {
             await this.underlyingPage.addScriptTag({ path: filePath });
         });
     }
+
+    public async withTracing<T>(wrappedFn: () => Promise<T>): Promise<T> {
+        return await withTracing(this.underlyingPage.tracing, wrappedFn);
+    }
+
+    private async screenshotOnError<T>(fn: () => Promise<T>): Promise<T> {
```

Comment:
should `fn` be named `wrappedFn` to be consisten with `withTracing` function signature?